[PyTorch] Hit fused addmm path in linear() for existing MHA

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #72871

We do this same trick in the native MHA implementation; backport it for purposes of fair comparison.

Differential Revision: [D34176090](https://our.internmc.facebook.com/intern/diff/D34176090/)

swolchok committed:
[PyTorch] Hit fused addmm path in linear() for existing MHA

We do this same trick in the native MHA implementation; backport it for purposes of fair comparison.

Differential Revision: [D34176090](https://our.internmc.facebook.com/intern/diff/D34176090/)

[ghstack-poisoned]

albanD reviewed (commented):
Shouldn't the linear function be doing that by default?

swolchok committed:
Update on "[PyTorch] Hit fused addmm path in linear() for existing MHA"


We do this same trick in the native MHA implementation; backport it for purposes of fair comparison.

Differential Revision: [D34176090](https://our.internmc.facebook.com/intern/diff/D34176090/)

[ghstack-poisoned]

swolchok commented:
> Shouldn't the linear function be doing that by default?

hmm, maybe I should try harder on https://github.com/pytorch/pytorch/pull/72728 first. Forgot about that one temporarily. Thanks!
